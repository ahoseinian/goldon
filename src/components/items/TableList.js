import React from 'react'
import Row from './Row'
import Icon from '../common/Icon'

const TableList = ({items}) => (
  <div className="table-responsive">
    <table className="table table-bordered">
      <thead>
        <tr>
          <th><Icon name="camera"/></th>
          <th>نام</th>
          <th>تعداد و قیمت</th>
          <th><Icon name="edit"/></th>
        </tr>
      </thead>
      <tbody>
        {items.map(x => <Row item={x} key={x.id}/>)}
      </tbody>
    </table>
  </div>
)

TableList.propTypes = {
  items: React.PropTypes.array.isRequired
}

export default TableList
