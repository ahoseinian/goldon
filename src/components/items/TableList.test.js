import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import TableList from './TableList'

describe('TableList', () => {
  it('renders', () => {
    const wrapper = shallow(<TableList items={[]}/>)
    expect(toJson(
      wrapper
    )).toMatchSnapshot()
  })
})
