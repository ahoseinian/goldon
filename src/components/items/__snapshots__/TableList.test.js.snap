exports[`TableList renders 1`] = `
<div
  className="table-responsive">
  <table
    className="table table-bordered">
    <thead>
      <tr>
        <th>
          <Icon
            name="camera" />
        </th>
        <th>
          نام
        </th>
        <th>
          تعداد و قیمت
        </th>
        <th>
          <Icon
            name="edit" />
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
